pageserver/src/tenant.rs

```diff
@@ -142,6 +142,9 @@ pub struct Tenant {
     /// Cached logical sizes updated updated on each [`Tenant::gather_size_inputs`].
     cached_logical_sizes: tokio::sync::Mutex<HashMap<(TimelineId, Lsn), u64>>,
     cached_synthetic_tenant_size: Arc<AtomicU64>,
+    /// Used by eviction tasks of this tenants timelines to synchronize calls to
+    /// `tenant::size::gather_inputs`.
+    eviction_task_synthetic_size: tokio::sync::Mutex<Option<Instant>>,
```

Comment:
NIT: `*_size: Option<Instant>` is quite confusing, maybe something like last_eviction_synthetic_size_warmup_at? It is quite long but describes the purpose

Reply:
The other merged PR changed this to be a named type instead: https://github.com/neondatabase/neon/pull/3877 so inclined to resolve.

Reply:
Only for timeline. Need the same for tenant

Reply:
Addressed in https://github.com/neondatabase/neon/pull/3871/commits/7e74ae2b94ba7bb64528b0069853e530397886e0